corehq/apps/export/views/list.py

```diff
@@ -43,58 +42,62 @@
 from corehq.privileges import EXCEL_DASHBOARD, DAILY_SAVED_EXPORT
 from corehq.util.download import get_download_response
 
-from corehq.apps.export.const import FORM_EXPORT, CASE_EXPORT, MAX_EXPORTABLE_ROWS, UNKNOWN_EXPORT_OWNER
+from corehq.apps.export.const import (
+    CASE_EXPORT,
+    FORM_EXPORT,
+    MAX_EXPORTABLE_ROWS,
+    UNKNOWN_EXPORT_OWNER,
+    SharingOption,
+)
 from corehq.apps.export.dbaccessors import (
-    get_form_export_instances,
     get_properly_wrapped_export_instance,
-    get_case_exports_by_domain,
-    get_form_exports_by_domain,
+    get_brief_case_exports_by_domain,
+    get_brief_form_exports_by_domain,
 )
 from corehq.apps.export.forms import CreateExportTagForm, DashboardFeedFilterForm
 from corehq.apps.export.models import FormExportInstance
 from corehq.apps.export.tasks import get_saved_export_task_status, rebuild_saved_export
 from corehq.apps.export.views.utils import ExportsPermissionsManager, user_can_view_deid_exports
 
 
-class BaseExportListView(HQJSONResponseMixin, BaseProjectDataView):
-    template_name = 'export/export_list.html'
-    allow_bulk_export = True
+class ExportListHelper(object):
+    '''
+    Encapsulates behavior that varies based on model (form, case, or sms)
+    and is needed by the function-based views in this module.
+    '''
+    form_or_case = None         # None if this handles both forms and cases
     is_deid = False
+    allow_bulk_export = True
 
-    lead_text = ugettext_lazy('''
-        Exports are a way to download data in a variety of formats (CSV, Excel, etc.)
-        for use in third-party data analysis tools.
-    ''')
-
-    @use_select2
-    @method_decorator(login_and_domain_required)
-    def dispatch(self, request, *args, **kwargs):
-        self.permissions = ExportsPermissionsManager(self.form_or_case, request.domain, request.couch_user)
-        self.permissions.access_list_exports_or_404(is_deid=self.is_deid)
-
-        return super(BaseExportListView, self).dispatch(request, *args, **kwargs)
+    @classmethod
+    def get(self, request, domain, form_or_case=None, is_daily_saved_export=False, is_feed=False, is_deid=False):
```

Comment:
It looks like the helper classes only needs `request` for `domain` and `couch_user` - you could drop it and pass those in directly, which would make this even more of a standalone helper.

Reply:
good point, I'll look at updating this code

Reply:
This isn't true (might have changed in a later commit): the helpers need access to `request.build_absolute_uri` and `is_location_safe_for_user` also needs to the request. I did add a commit to remove `domain` from the param list since it's already available.